src/App.jsx

```diff
@@ -1,247 +1,309 @@
-
-          <p className="panel-tabs has-text-weight-bold">
-            <a
-              data-cy="FilterAllUsers"
-              href="#/"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 1
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-              className="is-active"
-            >
-              User 2
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 3
-            </a>
-          </p>
-
-          <div className="panel-block">
-            <p className="control has-icons-left has-icons-right">
-              <input
-                data-cy="SearchField"
-                type="text"
-                className="input"
-                placeholder="Search"
-                value="qwe"
-              />
-
-              <span className="icon is-left">
-                <i className="fas fa-search" aria-hidden="true" />
-              </span>
-
-              <span className="icon is-right">
-                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-                <button
-                  data-cy="ClearButton"
-                  type="button"
-                  className="delete"
-                />
-              </span>
+import cn from 'classnames';
+
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+
+const products = productsFromServer.map((product) => {
+  const category = categoriesFromServer
+    .find(({ id }) => id === product.categoryId);
+
+  const user = usersFromServer.find(({ id }) => id === category.ownerId);
+
+  return {
+    id: product.id,
+    productName: product.name,
+    title: category.title,
+    icon: category.icon,
+    sex: user.sex,
+    userName: user.name,
+  };
+});
+
+const filterProduct = (userFilter, inputState, categoriesList) => {
+  if (!userFilter && !inputState && !categoriesList.length) {
+    return products;
+  }
+
+  let prods = [...products];
+
+  if (userFilter) {
+    prods = prods.filter(({ userName }) => userName === userFilter);
+  }
+
+  const newInputValue = inputState.trim().toLowerCase();
+
+  if (inputState) {
+    prods = prods.filter(({ productName }) => productName
+      .trim().toLowerCase().includes(newInputValue));
+  }
+
+  if (categoriesList.length) {
+    prods = prods.filter(({ title }) => categoriesList.includes(title));
+  }
+
+  return prods;
+};
+
+const sortProducts = (prevProds, sortState, sortCoefficient) => {
+  if (!sortCoefficient) {
+    return prevProds;
+  }
+
+  const copyProds = [...prevProds];
+
+  switch (sortState) {
+    case 'ID': return copyProds.sort((a, b) => (a.id - b.id) * sortCoefficient);
+    case 'Product': return copyProds
```

Comment:
you repeat the logic with repeating the array 3 times
will be better to sort array inside the switch case, and then return the final array only once at the end of the function